xtask/src/main.rs

```diff
@@ -108,6 +117,29 @@ fn options_for_chip(chip: Chip) -> Vec<Vec<String>> {
             .cloned()
             .collect::<Vec<_>>(),
         _ => default_options,
+    };
+    if !all_combinations {
+        return available_options;
+    } else {
+        // Return all the combination of availble options
+        let mut result = vec![];
+        for i in 0..(1 << available_options.len()) {
+            let mut options = vec![];
+            for j in 0..available_options.len() {
+                if i & (1 << j) != 0 {
+                    options.extend(available_options[j].clone());
+                }
+            }
+            result.push(options);
+        }
+        // Filter all the items that contains wifi and ble
+        let result = result
+            .into_iter()
+            .filter(|opts| {
+                !opts.contains(&"wifi".to_string()) || !opts.contains(&"ble".to_string())
```

Comment:
What's the purpose of this filter?

Reply:
Currently, when using ble and wifi, we throw an error: https://github.com/esp-rs/esp-generate/blob/main/src/main.rs#L408. I guess its related with https://github.com/esp-rs/esp-generate/issues/17. 

We should decide if we want to close the issue, and keep it like this. Or add support to allow using both options at the same time.

Reply:
We can fix this later, but we should definitely fix it as they aren't really mutually exclusive.

Probably worth fixing after lands: https://github.com/esp-rs/esp-hal/pull/2301 as it should simplify initializing things

Reply:
> We can fix this later, but we should definitely fix it as they aren't really mutually exclusive.

FYI, the TUI does not allow selecting both, this was a temporary fix to resolve #19. See [this comment](https://github.com/esp-rs/esp-generate/issues/19#issuecomment-2431121404).